clippy_lints/src/misc.rs

```diff
@@ -252,6 +263,10 @@ impl<'tcx> LateLintPass<'tcx> for LintPass {
             // Don't lint things expanded by #[derive(...)], etc or `await` desugaring
             return;
         }
+        // Do not emit if clippy::ref_pattern is not allowed to avoid having two lints for the same issue.
+        if !is_lint_allowed(cx, REF_PATTERN, expr.hir_id) {
+            return;
+        }
```

Comment:
This one appears to be above `used_underscore_binding` which would be unrelated

With that + a rebase/squash this looks good to merge to me!